Mathlib/Tactic/SolveByElim.lean

```diff
@@ -108,7 +108,28 @@ def mainGoalProc (cfg : Config := {}) (proc : MVarId → MetaM (List MVarId)) :
 
 /-- Create or modify a `Config` which calls `intro` on each goal before applying lemmas. -/
 def intros (cfg : Config := {}) : Config :=
-  mainGoalProc cfg fun g => do pure [(← g.intro1P).2]
+  cfg.mainGoalProc fun g => do pure [(← g.intro1P).2]
+
+/-- Attempt typeclass inference on each goal, before applying lemmas. -/
```

Comment:
Looking at the doc of `mainGoalProc` this seems to only do this on the main goal?

Reply:
I've added a comment explaining.